src/BitMapViewer.cpp

```diff
@@ -177,31 +177,25 @@ void BitMapViewer::refresh()
 void BitMapViewer::on_screenMode_currentIndexChanged(int index)
 {
 	screenMod = screenMode->itemText(index).toInt();
-	printf("\nnew screenMod: %i\n", screenMod);
-	//change then number of visibe pages when this changes
+	pageSize = screenMod >= 7 ? 65536 : 32768;
+
+	// change then number of visibe pages when this changes
 	imageWidget->setVramAddress(0);
+
 	// make sure that we are decoding new mode from page 0, imagine viewing
 	// page 3 of screen5 and then switching to screen 8 without changing the
 	// starting vram address....
 	imageWidget->setScreenMode(screenMod);
 	setPages();
 }
 
+static const char* pageNames[] = { "0", "1", "2", "3", "E0", "E1" };
+
 void BitMapViewer::setPages()
 {
-	int oldIndex = currentPage->currentIndex();
-
 	currentPage->clear();
-	currentPage->insertItem(0, "0");
-	currentPage->insertItem(1, "1");
-
-	if (screenMod < 7 && VDPDataStore::instance().getVRAMSize() > 0x10000) {
-		currentPage->insertItem(2, "2");
-		currentPage->insertItem(3, "3");
-	}
-
-	if (!useVDP && oldIndex < currentPage->count()) {
-		currentPage->setCurrentIndex(oldIndex);
+	for (unsigned int p = 0; p < VDPDataStore::instance().getVRAMSize() / pageSize; ++p) {
```

Comment:
Two details:
* In openMSX code we always write `unsigned` instead of `unsigned int`. I don't know what the habit is in the debugger code. And the same remark in BitMapViewer.h.
* No big deal, but this code repeatedly calculates `VDPDataStore::instance().getVRAMSize() / pageSize` (I didn't check how that's internally implemented, maybe it's cheap). But maybe calculate it just once in a local variable `pageCount`.